tests/checkpointing/test_legacy_checkpoints.py

```diff
@@ -14,84 +14,117 @@
 import glob
 import os
 import sys
+from unittest.mock import patch
 
 import pytest
+import torch
 
-from pytorch_lightning import Trainer
+import pytorch_lightning as pl
+from pytorch_lightning import Callback, Trainer
+from pytorch_lightning.callbacks import EarlyStopping
 from tests import _PATH_LEGACY
 
 LEGACY_CHECKPOINTS_PATH = os.path.join(_PATH_LEGACY, 'checkpoints')
 CHECKPOINT_EXTENSION = ".ckpt"
+LEGACY_BACK_COMPATIBLE_PL_VERSIONS = (
+    "1.0.0",
+    "1.0.1",
+    "1.0.2",
+    "1.0.3",
+    "1.0.4",
+    "1.0.5",
+    "1.0.6",
+    "1.0.7",
+    "1.0.8",
+    "1.1.0",
+    "1.1.1",
+    "1.1.2",
+    "1.1.3",
+    "1.1.4",
+    "1.1.5",
+    "1.1.6",
+    "1.1.7",
+    "1.1.8",
+    "1.2.0",
+    "1.2.1",
+    "1.2.2",
+    "1.2.3",
+    "1.2.4",
+    "1.2.5",
+    "1.2.6",
+    "1.2.7",
+    "1.2.8",
+    "1.2.10",
+    "1.3.0",
+    "1.3.1",
+    "1.3.2",
+    "1.3.3",
+    "1.3.4",
+    "1.3.5",
+    "1.3.6",
+    "1.3.7",
+    "1.3.8",
+)
 
 
-# todo: add more legacy checkpoints - for < v0.8
-@pytest.mark.parametrize(
-    "pl_version",
-    [
-        # "0.8.1",
-        "0.8.3",
-        "0.8.4",
-        # "0.8.5", # this version has problem with loading on PT<=1.4 as it seems to be archive
-        # "0.9.0", # this version has problem with loading on PT<=1.4 as it seems to be archive
-        "0.10.0",
-        "1.0.0",
-        "1.0.1",
-        "1.0.2",
-        "1.0.3",
-        "1.0.4",
-        "1.0.5",
-        "1.0.6",
-        "1.0.7",
-        "1.0.8",
-        "1.1.0",
-        "1.1.1",
-        "1.1.2",
-        "1.1.3",
-        "1.1.4",
-        "1.1.5",
-        "1.1.6",
-        "1.1.7",
-        "1.1.8",
-        "1.2.0",
-        "1.2.1",
-        "1.2.2",
-        "1.2.3",
-        "1.2.4",
-        "1.2.5",
-        "1.2.6",
-        "1.2.7",
-        "1.2.8",
-        "1.2.10",
-        "1.3.0",
-        "1.3.1",
-        "1.3.2",
-        "1.3.3",
-        "1.3.4",
-        "1.3.5",
-        "1.3.6",
-        "1.3.7",
-        "1.3.8",
-    ]
-)
-def test_resume_legacy_checkpoints(tmpdir, pl_version: str):
-    path_dir = os.path.join(LEGACY_CHECKPOINTS_PATH, pl_version)
+@pytest.mark.parametrize("pl_version", LEGACY_BACK_COMPATIBLE_PL_VERSIONS)
+def test_load_legacy_checkpoints(tmpdir, pl_version: str):
```

Comment:
What is the difference supposed to be between the tests

Reply:
One is load from checkpoint other is resume, or what are you asking?

Reply:
But do you expect there to be a difference between them?
`resume_from_checkpoint` loads from a checkpoint

Reply:
shall not, but still rather test it, then just hope :]